jbmc/unit/java_bytecode/java_replace_nondet/replace_nondet.cpp

```diff
@@ -110,26 +145,54 @@ void load_and_test_method(
   remove_virtual_functions(model_function);
 
   // Then test both situations.
-  THEN(
-    "Code should work when remove returns is called before "
-    "replace_java_nondet.")
+  THEN("Replace nondet should work when remove returns has been called.")
   {
     remove_returns(model_function, [](const irep_idt &) { return false; });
 
     replace_java_nondet(model_function);
 
-    validate_method_removal(goto_function.body.instructions);
+    validate_nondet_method_removed(goto_function.body.instructions);
+  }
+
+  THEN("Replace nondet should work when remove returns hasn't been called.")
+  {
+    replace_java_nondet(model_function);
+
+    remove_returns(model_function, [](const irep_idt &) { return false; });
+
+    validate_nondet_method_removed(goto_function.body.instructions);
+  }
+
+  object_factory_parameterst params{};
+
+  THEN(
+    "Replace and convert nondet should work when remove returns has been "
+    "called.")
+  {
+    replace_java_nondet(model_function);
+
+    convert_nondet(model_function, null_message_handler, params, ID_java);
+
+    remove_returns(model_function, [](const irep_idt &) { return false; });
+
+    std::stringstream out;
+    goto_function.body.output(namespacet(symbol_table), "", out);
+    std::string res = out.str();
+
+    validate_nondets_converted(goto_function.body.instructions);
   }
 
   THEN(
-    "Code should work when remove returns is called after "
-    "replace_java_nondet.")
+    "Replace and convert nondet should work when remove returns hasn't been "
+    "called.")
   {
     replace_java_nondet(model_function);
 
+    convert_nondet(model_function, null_message_handler, params, ID_java);
+
     remove_returns(model_function, [](const irep_idt &) { return false; });
```

Comment:
Still called despite text in `THEN`

Reply:
I have changed the text to make it clear it's about "before" or "after" `remove_returns` is run.